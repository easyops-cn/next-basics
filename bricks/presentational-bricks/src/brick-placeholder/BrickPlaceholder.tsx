import React from "react";

import style from "./brick-placeholder.module.css";

interface BrickPlaceholderProps {
  text?: string;
}

export function BrickPlaceholder(
  props: BrickPlaceholderProps
): React.ReactElement {
  return <div className={style.placeholder}>{props.text}</div>;
}
