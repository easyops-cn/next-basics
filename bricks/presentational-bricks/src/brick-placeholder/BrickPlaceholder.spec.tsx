import React from "react";
import { render } from "@testing-library/react";

import { BrickPlaceholder } from "./BrickPlaceholder";

describe("BrickPlaceholder", () => {
  it("should work", () => {
    const result = render(<BrickPlaceholder text="" />);
    const asFragment = result.asFragment;
    expect(asFragment()).toMatchSnapshot();
  });
});
