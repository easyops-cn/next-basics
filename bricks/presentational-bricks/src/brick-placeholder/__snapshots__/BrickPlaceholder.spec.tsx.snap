// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrickPlaceholder should work 1`] = `
<DocumentFragment>
  <div
    class="placeholder"
  />
</DocumentFragment>
`;
